### DISCUSSION TITLE
blast検索について

### DISCUSSION THREAD

**Discussion by yc11mh22** [nig-sc/nigsc_homepage2#471] [Q&A]
#!/bin/sh
#$ -S /bin/bash
#$ -cwd
#$ -N blastxbatch

singularity exec --bind /usr/local/seq /usr/local/biotools/b/blast:2.9.0--pl526h979a64d_3 blastp -db /usr/local/seq/blast/ncbi/nr -query query.fasta -outfmt 6 -evalue 1e-03 > blastpcutoff1e-3_query.fasta.csv

以上のスクリプトでjobを行ったのですが、

以下のようなエラーが出てうまくいきません。

ERROR: Unable to locate a modulefile for 'gcc'
WARNING: Skipping mount /opt/pkg/singularity-ce/4.0.0/var/singularity/mnt/session/etc/resolv.conf [files]: /etc/resolv.conf doesn't exist in container
Error: mdb_env_open: Cannot allocate memory

以前はこのようなスクリプトで問題なく動いていました。
どなたか解決方法がわかる方いたら助けてください。

**Comment by sc-fs** [comment]
「ERROR: Unable to locate a modulefile for 'gcc'」
こちらは以下を参照して、bashrcの修正をお願いします。
https://sc.ddbj.nig.ac.jp/faq/faq_os_migration_login/#module_load

「WARNING: Skipping mount /opt/pkg/singularity-ce/4.0.0/var/singularity/mnt/session/etc/resolv.conf [files]: /etc/resolv.conf doesn't exist in container」
これらは動作に影響ないので、一旦無視してください。

「Error: mdb_env_open: Cannot allocate memory」
こちらは動作時のメモリが不足している可能性がありますので、
スクリプトの先頭でs_vmem,mem_reqにてメモリ指定をしてください。
#!/bin/sh
#$ -S /bin/bash
#$ -cwd
#$ -N blastxbatch
#$ -l s_vmem=100G
#$ -l mem_req=100G
